cmd/kn/main.go

```diff
@@ -64,21 +64,26 @@ func run(args []string) error {
 		return err
 	}
 
-	// Strip of all flags to get the non-flag commands only
-	commands, err := stripFlags(args)
+	pluginManager := plugin.NewManager(config.GlobalConfig.PluginsDir(), config.GlobalConfig.LookupPluginsInPath())
+
+	// Create kn root command and all sub-commands
+	rootCmd, err := root.NewRootCommand(pluginManager.HelpTemplateFuncs())
 	if err != nil {
 		return err
 	}
 
-	// Find plugin with the commands arguments
-	pluginManager := plugin.NewManager(config.GlobalConfig.PluginsDir(), config.GlobalConfig.LookupPluginsInPath())
-	plugin, err := pluginManager.FindPlugin(commands)
+	// temporary setting to parse all flags
+	rootCmd.FParseErrWhitelist = cobra.FParseErrWhitelist{UnknownFlags: true}
+	// Strip of all flags to get the non-flag commands only
+	commands, err := stripFlags(rootCmd, args)
 	if err != nil {
 		return err
 	}
```

Comment:
I can comment and all tests pass. Since `stripFlags` is added by you, perhaps a test is warranted?

Reply:
`stripFlags` been around, this PR changes the implementation of it, the function explicitly takes argument with (temporary) settings added on the command as `rootCmd.FParseErrWhitelist = cobra.FParseErrWhitelist{UnknownFlags: true}` to allow passing unknown flags (unknown to core kn, to provision plugins). I will try and add a case to cover this. thanks!

Reply:
Added a unit test without aforementioned temp setting on command under test.